designs/cloudWatch/cloudWatchIntegration.md

```diff
@@ -0,0 +1,77 @@
+# CloudWatch Logs Experience
+
+* **Type**: UX Design
+* **Status**: Proposal
+
+## Introduction
+CloudWatch Logs enables a user to centralize the logs from all of their systems, applications, and AWS services that they use, in a single, highly scalable service. This document describes adding CloudWatch Logs functionality into the JetBrains toolkit in order to facilitate easily viewing and searching them for specific error codes or patterns.
+
+## Comparison to other products
+
+### AWS Toolkit for VSCode  
+Not supported
+
+### AWS Toolkit for Visual Studio  
+Supports listing of Lambda CloudWatch Log Streams and downloading them to a temporary file and opening it in Notepad
+
+## Experience
+
+### View Log Groups
+The AWS Explorer gains a CloudWatch root node for interacting wit CloudWatch service. The node lists all the Log Groups that exist in the selected region for the selected credential profile as child nodes. Double clicking a Log Group opens the CloudWatch tool window with the tab for the Log Streams that belong to that group. A new tab is in the tool window is created if one does not already exist.
+
+*Browsing Log Groups in AWS Explorer:*<br />
+![BROWSE_LOG_GROUP]
+
+#### Alternatives Considered
+Instead of building a new CloudWatch service node in the AWS Explorer, we instead build a CloudWatch Log Group browser tab in the CloudWatch tool window. This new tab is also be linked to the active AWS connection settings. This differs from the other tabs which may lead to a confusing user experience and seen as a con.
+
+*Possible alternative for browsing Log Groups:*<br />
+![ALT_BROWSE_LOG_GROUP]
+
+### View Log Streams
+Upon selecting a Log Group, a new tab in the CloudWatch tool window is shown to the user. This new tab lists all the streams for the requested group to allow the user to find the exact stream they are looking for.
+
+Log Streams can also be sorted by name or last event time in order to help facilitate finding the Log Stream of interest.
+
+*View all Log Streams in the selected Log Group:*<br />
+![BROWSE_LOG_STREAMS]
+
+#### Navigate to a Log Stream
+Double clicking the Log group in the AWS Explorer opens the Log Stream selection. 
+
+For additional service integrations, such as a Lambda function, where we know what Log Group is associated with the resource a new context menu action adds the ability to bypass having to search for the Log Group and instead open the Log Stream selection directly.
+
+### View A Log Stream
```

Comment:
What happens if new logs come in after the user loads the log stream. Does the view refresh automatically? Or is there a button to do this manually?

Reply:
In non-tail (live?) mode, we would probably operate like the console and when they reach the end have a Check for More row similar to the S3 truncation row.